### ISSUE TITLE
implementation of Airbnb's accessors-boolean-prefix

### ISSUE THREAD

**Issue by aervin** [palantir/tslint#3620]
I'd like a rule for enforcing the boolean-prefix guideline from the [Airbnb style guide](https://github.com/airbnb/javascript#accessors--boolean-prefix). I don't think `variable-name` is set up for this, but that's where I've looked for this capability. 

Would throw on the Customer class method `age`
```ts
Customer.age();
```
in the event that the `age` method returns a boolean. Could be corrected by prefixing with one of a selection of words ("is", "has", "could", "should"?), so
```ts
Customer.hasAge(); 
```

* What about unions with booleans?
* Why stop at class members?

**Comment by ebrpol** [comment]
Did you find another way to use such a rule?
